esp_idf_monitor/base/serial_reader.py

```diff
@@ -125,7 +127,10 @@ def set_closing_wait(self, delay_sec):  # type: (int) -> None
 
         # set serial_struct
         buf = struct.pack(struct_format, *serial_struct)
-        fcntl.ioctl(self.serial.fd, termios.TIOCSSERIAL, buf)
+        try:
+            fcntl.ioctl(self.serial.fd, termios.TIOCSSERIAL, buf)
+        except PermissionError:
+            pass
```

Comment:
How about using the tcflush here in except? This way we would still try to wait for the data to be sent if you had permission to set closing_wait, but would also mean that if you don't have the permission console won't get stuck. What do you think?

The way it is implemented currently, it makes no sense to set the closing_wait as based on documentation tcflush will discard all data waiting for writing. The closing_wait was set to 1 second to at least try to finish writing while at the same time making sure the console won't get stuck for a noticeable time. We need to make sure it does work as well for some slower systems that might benefit from the longer timeout.

Reply:
The value used for `closing_wait` is irrelevant because closing the serial port has no impact on whether or not `cdc_acm` will continue to transmit previously written characters. It's just a thing the tty layer does that waits for all data to be transmitted before letting close() finish. The data will still be written to the serial port. I will change it to disable the timeout instead of waiting unnecessarily for 1 second.

This is supposed to be an interactive application. Sending characters and then expecting them to have all been received without waiting for a response shouldn't need to be supported? Everything is being sent 1 character at a time and `cdc_acm` only has 16 transmit buffers so this is limited to a queue of only 16 characters.

I'm going to rework it a bit because I've made another kernel patch that makes it easier to fix the `closing_wait` time using udev rules. We only need to discard data if the timeout hasn't been changed.

Reply:
I've included a 300ms delay before discarding data to match your other 300ms delay in `write()`.